naptime-graphql/src/main/scala/org/coursera/naptime/ari/graphql/schema/NaptimePaginatedResourceField.scala

```diff
@@ -173,9 +173,8 @@ object NaptimePaginatedResourceField {
             idsWithStart.take(limit)
           }.getOrElse(List.empty)
         }
-        objects.collect {
-          case (id, element) if ids.contains(id) => element
-        }.toSeq
+        val objectMap = objects.map { case (id, element) => id -> element }.toMap
+        ids.map(id => objectMap(id))
```

Comment:
i believe this could throw an exception if we somehow have an id that we don't have an object for. I'd recommend making it safer by doing:
```scala
ids.flatMap(id => objectMap.get(id))
```